src/ints/xms.cpp

```diff
@@ -16,478 +17,843 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
  */
 
-
-#include <stdlib.h>
-#include <string.h>
-#include <stddef.h>
 #include "dosbox.h"
+
+#include "bios.h"
+#include "bitops.h"
 #include "callback.h"
+#include "checks.h"
+#include "cpu.h"
+#include "dos_inc.h"
+#include "inout.h"
 #include "mem.h"
 #include "regs.h"
-#include "dos_inc.h"
 #include "setup.h"
-#include "inout.h"
-#include "xms.h"
-#include "bios.h"
+#include "support.h"
 
-#define XMS_HANDLES							50		/* 50 XMS Memory Blocks */ 
-#define XMS_VERSION    						0x0300	/* version 3.00 */
-#define XMS_DRIVER_VERSION					0x0301	/* my driver version 3.01 */
-
-#define	XMS_GET_VERSION						0x00
-#define	XMS_ALLOCATE_HIGH_MEMORY			0x01
-#define	XMS_FREE_HIGH_MEMORY				0x02
-#define	XMS_GLOBAL_ENABLE_A20				0x03
-#define	XMS_GLOBAL_DISABLE_A20				0x04
-#define	XMS_LOCAL_ENABLE_A20				0x05
-#define	XMS_LOCAL_DISABLE_A20				0x06
-#define	XMS_QUERY_A20						0x07
-#define	XMS_QUERY_FREE_EXTENDED_MEMORY		0x08
-#define	XMS_ALLOCATE_EXTENDED_MEMORY		0x09
-#define	XMS_FREE_EXTENDED_MEMORY			0x0a
-#define	XMS_MOVE_EXTENDED_MEMORY_BLOCK		0x0b
-#define	XMS_LOCK_EXTENDED_MEMORY_BLOCK		0x0c
-#define	XMS_UNLOCK_EXTENDED_MEMORY_BLOCK	0x0d
-#define	XMS_GET_EMB_HANDLE_INFORMATION		0x0e
-#define	XMS_RESIZE_EXTENDED_MEMORY_BLOCK	0x0f
-#define	XMS_ALLOCATE_UMB					0x10
-#define	XMS_DEALLOCATE_UMB					0x11
-#define XMS_QUERY_ANY_FREE_MEMORY			0x88
-#define XMS_ALLOCATE_ANY_MEMORY				0x89
-#define	XMS_GET_EMB_HANDLE_INFORMATION_EXT	0x8e
-#define XMS_RESIZE_ANY_EXTENDED_MEMORY_BLOCK 0x8f
-
-#define	XMS_FUNCTION_NOT_IMPLEMENTED		0x80
-#define	HIGH_MEMORY_NOT_EXIST				0x90
-#define	HIGH_MEMORY_IN_USE					0x91
-#define	HIGH_MEMORY_NOT_ALLOCATED			0x93
-#define XMS_OUT_OF_SPACE					0xa0
-#define XMS_OUT_OF_HANDLES					0xa1
-#define XMS_INVALID_HANDLE					0xa2
-#define XMS_INVALID_SOURCE_HANDLE			0xa3
-#define XMS_INVALID_SOURCE_OFFSET			0xa4
-#define XMS_INVALID_DEST_HANDLE				0xa5
-#define XMS_INVALID_DEST_OFFSET				0xa6
-#define XMS_INVALID_LENGTH					0xa7
-#define XMS_BLOCK_NOT_LOCKED				0xaa
-#define XMS_BLOCK_LOCKED					0xab
-#define	UMB_ONLY_SMALLER_BLOCK				0xb0
-#define	UMB_NO_BLOCKS_AVAILABLE				0xb1
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
+
+CHECK_NARROWING();
+
+// ***************************************************************************
+// Constants and type definitions
+// ***************************************************************************
+
+constexpr uint16_t XMS_VERSION        = 0x0300; // version 3.00
+constexpr uint16_t XMS_DRIVER_VERSION = 0x0301; // driver version 3.01
+
+// MS-DOS 6.22 defaults to 32 XMS handles, we can provide more without any
+// significant cost.
+constexpr uint8_t NumXmsHandles = 128;
+
+enum class Result : uint8_t {
```

Comment:
:+1: , nice to get this namespace cleaned up!

Reply:
This is so much more readable, knowing that all the majority of these XMS functions are actually returning specific `Result` codes instead of their previous `Bitu` quantities.